pkgs/development/libraries/x264/default.nix

```
@@ -28,6 +28,10 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     patchShebangs .
+  ''
+  # Darwin uses `llvm-strip`, which results in a crash at runtime in assembly-based routines when `-x` is specified.
+  + lib.optionalString stdenv.isDarwin ''
```

Does the clangStdenv generally do this? If so, perhaps this would be more appropriate

```suggestion
  + lib.optionalString stdenv.isClang ''
```

---

clangStdenv uses GNU Binutils on Linux. It would need to be `stdenv.cc.bintools.isLLVM` to check for LLVM, but that won’t be true on Darwin (where it’s neither), so you’d still also need `stdenv.isDarwin`.

I tried building pkgsLLVM.ffmpeg on Linux from the 23.11 release branch, but it failed building Berkeley DB, so I don’t know whether it would pass the test. However, I do know cctools is more conservative than LLVM.

I assume there’s something about the way the assembly is written that breaks on Mach-O when `llvm-strip -x` is run. My initial fix used `strip` from cctools, and that worked as well. I only tried this approach after that since it worked with rav1e and allows Darwin to avoid a dependency on cctools when it doesn’t need it.